extensions/podman/src/extension.ts

```diff
@@ -367,6 +368,39 @@ export async function checkDefaultMachine(machines: MachineJSON[]): Promise<void
     }
   }
 
+  // check if connection is in sync with machine. If the default connection is rootless but the machine is rootful ask the user to update the connection
+  if (defaultConnectionNotify && !!runningMachine?.Default) {
+    const defaultConnection = await getDefaultConnection();
+    const isRootful = await isRootfulMachine(runningMachine.Name);
+    if (!defaultConnection?.Name.endsWith(ROOTFUL_SUFFIX) && isRootful) {
```

Comment:
I don't get it you are comparing the default connection with a non default machine ?

Reply:
In the if above i check that the running machine is the default one